src/IMirrorNodeResponses.sol

```diff
@@ -39,4 +39,14 @@ interface IMirrorNodeResponses {
         int64 amount;
         string denominating_token_id;
     }
+
+    struct TokenRelationship {
```

Comment:
Q: Does this represent a Token entity or the mapping of ownership by an account holding a token?

Reply:
It represents the response of this endpoint https://testnet.mirrornode.hedera.com/api/v1/docs/#/accounts/getTokensByAccountId from the Mirror Node (without the enclosing array because we provide the `token.id` when querying the Mirror Node).

Reply:
Yes, this is the `TokenRelationship` response object described in the schema for: https://testnet.mirrornode.hedera.com/api/v1/docs/#/accounts/getTokensByAccountId

```solidity
if (vm.keyExistsJson(json, ".tokens")) {
  bytes memory tokens = vm.parseJson(json, ".tokens");
  IMirrorNodeResponses.TokenRelationship[] memory relationships = abi.decode(tokens, (IMirrorNodeResponses.TokenRelationship[]));
  return relationships.length > 0;
}
```
- If the `".tokens"` field exists:
  - we try to parse the array of token relationships
  - since we have passed a `token.id` param with the token ID, if the array is not empty it means that the given token is associated with the account and we return `true`
- If the `".tokens"` field doesn't exist:
  - most likely Mirror Node returned `{"_status":{"messages":[{"message":"Not found"}]}}`, which means the given account address doesn't exist
  - in this case, we just ignore the error and return a default response of `false`